game/hud/src/widgets/Crafting/components/NameInput/index.tsx

```diff
@@ -0,0 +1,48 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ *
+ * @Author: Mehuge (mehuge@sorcerer.co.uk)
+ * @Date: 2017-05-13 18:19:58
+ * @Last Modified by: Mehuge (mehuge@sorcerer.co.uk)
+ * @Last Modified time: 2017-05-24 19:55:40
+ */
+
+import * as React from 'react';
+import { connect } from 'react-redux';
+import Select from '../Select';
+import Label from '../Label';
+import Input from '../Input';
+import { GlobalState } from '../../services/session/reducer';
+import { StyleSheet, css, merge, nameInput, NameInputStyles } from '../../styles';
+
+export interface NameInputReduxProps {
+  dispatch?: (action: any) => void;
+  name?: string;
+  style?: Partial<NameInputStyles>;
+}
+
+export interface NameInputProps extends NameInputReduxProps {
+  onChange: (name: String) => void;
+}
+
+interface NameInputState {}
+
+const select = (state: GlobalState, props: NameInputProps) : NameInputReduxProps => {
+  return {
+    name: state.job.name,
+  };
+};
+
+const NameInput = (props: NameInputProps) => {
+  const ss = StyleSheet.create(merge({}, nameInput, props.style));
+  return (
+    <div className={'name-input ' + css(ss.container)}>
```

Comment:
Well, I think the reason we have a generic container name is so that if we are passing in overrides `style={{ container: some.overrideStyles }}` we know the key we need to override is called container.

We would need to change that standard (which is throughout the codebase) and the component consumer would need documentation to tell them to override styles they need to use the key: `nameInput` or `nameInputContainer` (the latter I think is too verbose).